### PR TITLE
Routing and forwarding to login in case of invalid token

### PR DESCRIPTION
**Pull request by stedo005**

### CODE REVIEW

File: frontend/src/Main/Categories.tsx [hunk 1]
```diff
@@ -55,6 +60,7 @@ const Categories = () => {
             .then(() => setCategoryName(""))
             .then(fetchCategories)
             .catch((e: Error) => {
+                navigate("../login")
```
Comment (LetterBe) [inline frontend/src/Main/Categories.tsx:63]: for the navigate is it really necessary the ('../login') ? in my code I only have ('/login) and it works

File: frontend/src/Main/Categories.tsx [hunk 2]
```diff
@@ -81,7 +89,11 @@ const Categories = () => {
                 "Authorization": "Bearer " + localStorage.getItem("token")
```
Comment (LetterBe) [inline frontend/src/Main/Categories.tsx:89]: why do you save your token here?

File: frontend/src/Main/Overview.tsx [hunk 3]
```diff
@@ -16,27 +19,33 @@ const Overview = () => {
...
         })
+            .then(response =>
+                checkLogin(response)
+            )
             .then(getLastUpdate)
+            .catch(() => navigate("../login"))
     }
 
-    const getLastUpdate = () => {
+    const getLastUpdate = useCallback(() => {
```
Comment (LetterBe) [inline frontend/src/Main/Overview.tsx:29]: for me: why in this function you are now using a callBack and before you werent?

**Review by LetterBe — COMMENTED** [review]
I made some comments and asked some questions, but you need someone to approve, okay?

File: frontend/src/Main/Categories.tsx (hunk 1 above)
Comment (stedo005) [reply frontend/src/Main/Categories.tsx:63]: it is not necessary, i do this for habit

File: frontend/src/Main/Categories.tsx (hunk 2 above)
Comment (stedo005) [reply frontend/src/Main/Categories.tsx:89]: In this case i get the token from localStorage. Setting is on login.

File: frontend/src/Main/Overview.tsx (hunk 3 above)
Comment (stedo005) [reply frontend/src/Main/Overview.tsx:29]: I added a useEffect, and without the callback there is an infinity loop. In the terminal react shows you the warnings.

**Review by LetterBe — APPROVED** [review]
looks nice :D

**Pull request merged**